docs/extensions_preview.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Extensions Preview
+layout: docs
+---
+
+Introducing Tilt Extensions! Extensions are a simple way to bundle up Tiltfile functionality that should be shared with the world.
```

Comment:
This opening is very focused on the perspective of the extension author, rather than the extension user, which maybe raises the question of what's the goal of this?

Some potential answers:
1. convince people to write extensions
2. convince people to go check out extensions and make use of them
3. just generally make people optimistic about Tilt's flexibility and momentum

This is just a gut feeling, but I suspect even for (1), it'd be better motivation to write this from the extension user's perspective "look, someone wrote this once and now everyone can use and learn from it!". Which maybe all the like "yeah well that's what writing code and open source is good for" might be obvious when you give it a second's thought, but it still might be worth discussing from that angle instead to make sure it's top-of-mind?

I think it also feels closer to how I would phrase the root use case of extensions, along the lines of "as a Tiltfile author, I would like a richer set of pre-written functionality available to me"